### DISCUSSION TITLE
How to update a document which contains array of another Schema in mongoose?

### DISCUSSION THREAD

**Discussion by polaris404** [cesanta/mongoose#2122] [Ask questions, get answers] [answered]
I have Task Schema and User Schema in one file

    const mongoose = require("mongoose");
    const validator = require("validator");
    
    const TaskSchema = new mongoose.Schema({
    	name: {
    		type: String,
    		required: [true, "must provide name"],
    		trim: true,
    		maxlength: [20, "name can not be more than 20 characters"],
    	},
    	completed: {
    		type: Boolean,
    		default: false,
    	},
    	createdAt: Date,
    	updatedAt: Date,
    	completedAt: Date,
    });
    
    const UserSchema = new mongoose.Schema({
    	name: {
    		type: String,
    		required: [true, "Must provide name"],
    		trim: true,
    		maxlength: [20, "Name cannot be more than 20 characters"],
    	},
    	email: {
    		type: String,
    		unique: true,
    		required: [true, "Must provide email"],
    		validator: {
    			validator: validator.isEmail,
    			message: "Please provide valid email",
    		},
    	},
    	pass: {
    		type: String,
    		required: [true, "Must provide password"],
    	},
    	verificationToken: String,
    	isVerified: {
    		type: Boolean,
    		default: false,
    	},
    	verifiedAt: Date,
    	tasks: {
    		type: [TaskSchema],
    	},
    });
    
    module.exports = mongoose.model("User", UserSchema);

Now I want to update/insert task in `tasks` array inside UserSchema. So how do I update a document with a document of different Schema. I'm only storing Users in database, not Tasks. Also is this the correct way of defining array of TaskSchema in UserSchema?

I searched a lot but couldn't get an proper answer.
Thank You

**Comment by scaprile** [comment, marked as answer]
Please read any docs or description or even the README in this repo and realize that this is a Network Library, not even remotely related in any way to a database.
